### PR TITLE
cleanup unused methods and endpoints

### PR DESCRIPTION
**Pull request by nulls**
First part of changes from #1105: cleanup of `CloneRepositoryController.kt` and support changes in `save-cli`

### CODE REVIEW

**Comment by nulls** [comment]
Save CLI part is refactored without testing, it will be covered in scope #1061

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/SelectForm.kt [hunk 1]
```diff
@@ -109,6 +109,7 @@ fun <D : Any> selectFormRequired() = FC<SelectFormRequiredProps<D>> { props ->
                 className = ClassName("form-control")
                 id = props.formType.name
                 required = true
+                // TODO: why we need an extra option
```
Comment (petertrr) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/SelectForm.kt:112]: I believe it's an empty option which is selected by default

File: save-api/src/main/kotlin/com/saveourtool/save/api/utils/RequestUtils.kt [hunk 2]
```diff
@@ -101,60 +84,17 @@ suspend fun HttpClient.uploadAdditionalFile(
...
-                    json.encodeToString(shortFileInfo),
-                    Headers.build {
-                        append(HttpHeaders.ContentType, ContentType.Application.Json)
-                    }
-                )
-            }
-        }))
-    }
+suspend fun HttpClient.submitExecution(runExecutionRequest: RunExecutionRequest): HttpResponse = this.post {
+    url("${Backend.url}/api/$v1/run/trigger")
```
Comment (petertrr) [inline save-api/src/main/kotlin/com/saveourtool/save/api/utils/RequestUtils.kt:94]: I believe this should contain `testingType` as a request param?

Comment (nulls) [reply save-api/src/main/kotlin/com/saveourtool/save/api/utils/RequestUtils.kt:94]: yes

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/SelectForm.kt (hunk 1 above)
Comment (nulls) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/SelectForm.kt:112]: probably, but there is `value = props.selectedValue`. maybe we need to add an empty value only if `props.selectedValue` is not presented in `elements`. Need to discuss with @sanyavertolet

File: save-api/src/main/kotlin/com/saveourtool/save/api/utils/RequestUtils.kt (hunk 2 above)
Comment (nulls) [reply save-api/src/main/kotlin/com/saveourtool/save/api/utils/RequestUtils.kt]: returned back

File: save-api/src/main/kotlin/com/saveourtool/save/api/SaveCloudClient.kt [hunk 3]
```diff
@@ -81,126 +77,50 @@ class SaveCloudClient(
...
+    ): RunExecutionRequest? {
+        val runExecutionRequest = RunExecutionRequest(
+            projectCoordinates = ProjectCoordinates(
+                organizationName = evaluatedToolProperties.organizationName,
+                projectName = evaluatedToolProperties.projectName,
+            ),
+            testSuiteIds = evaluatedToolProperties.testSuites
+                .split(DATABASE_DELIMITER)
+                .map { it.toLong() },
+            files = additionalFiles?.map { it.toStorageKey() }.orEmpty(),
```
[the remark this answers is not in the document]
Comment (nulls) [reply save-api/src/main/kotlin/com/saveourtool/save/api/SaveCloudClient.kt:96]: it's used in backend. we put files to `RunExecutionRequest` and then it's put to `Execution`

**Review by petertrr — APPROVED** [review]

**Pull request merged**